### ISSUE TITLE
Build issue on mono 4 

### ISSUE THREAD

**Issue by julien-leclercq** [fsharp/FSharp.Data#962]
Step to reproduce : 

1/ try to build on mono 4 get this 

```
  1) Building /home/julien/Documents/programming/fsharp/FSharp.Data/FSharp.Data.sln failed with exitcode 1.
  2) : /usr/lib/mono/4.5/Microsoft.Common.targets(0,0): PCL Reference Assemblies not installed.
  3) : /usr/lib/mono/4.5/Microsoft.Common.targets(0,0): PCL Reference Assemblies not installed.
  4) : /usr/lib/mono/4.5/Microsoft.Common.targets(0,0): PCL Reference Assemblies not installed.
```

after this i tried to install the referenceassemblies-pcl package (i am on ubuntu 14.04) 
and then I get this errors 

```
1) Building /home/julien/Documents/programming/fsharp/FSharp.Data/FSharp.Data.sln failed with exitcode 1.
  2) FS0219: FSC(0,0): The referenced or default base CLI library 'mscorlib' is binary-incompatible with the referenced F# core library '/usr/lib/mono/gac/FSharp.Core/4.4.0.0__b03f5f7f11d50a3a/FSharp.Core.dll'. Consider recompiling the library or making an explicit reference to a version of this library that matches the CLI version you are using.
  3) FS0218: FSC(0,0): Unable to read assembly '/usr/lib/mono/gac/FSharp.Core/4.4.0.0__b03f5f7f11d50a3a/FSharp.Core.dll'
  4) FS0193: FSC(0,0): The module/namespace 'System.Net' from compilation unit 'System' did not contain the namespace, module or type 'WebClient'
  5) FS0193: FSC(0,0): The module/namespace 'System.Net' from compilation unit 'System' did not contain the namespace, module or type 'WebClient'
```

**Comment by jschiefer** [comment]
Seeing the same thing, Ubuntu 14.04 also.
``` 
mono --version
Mono JIT compiler version 4.4.1 (Nightly 4.4.1.0/4747417 Wed Jun 22 11:14:49 UTC 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        LLVM:          supported, not enabled.
        GC:            sgen
```

**Comment by jaegerpicker** [comment]
I get the same error on a Mac. 

Mono JIT compiler version 4.2.2 (Stable 4.2.2.30/996df3c Wed Feb 10 13:36:17 PST 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	LLVM:          supported, not enabled.
	GC:            sgen

**Comment by jschiefer** [comment]
I did a few experiments with Travis-CI, building with different versions of Mono. I started from the last commit that passed the CI on master(b26c1ca47705bb6800f2db5ff111f28678d7dfdd), and tried building with different versions of Mono by changing the version in .travis.yml (this builds on OSX BTW).

Results:
- 4.2.3 Pass
- 4.2.4 Pass
- 4.3.2 Fail
- 4.4.1 Fail

So it appears that this issue was introduced with Mono 4.3 (although this is inconsistent with what @jaegerpicker observed).

The Mono 4.4 release notes mention [some changes to building with reference assemblies](http://www.mono-project.com/docs/about-mono/releases/4.4.0/#class-libraries) that I cannot claim to understand at this point, but they seem related.
